Source/WebInspectorUI/UserInterface/Controllers/JavaScriptRuntimeCompletionProvider.js

```diff
@@ -383,6 +383,13 @@ WI.JavaScriptRuntimeCompletionProvider = class JavaScriptRuntimeCompletionProvid
                 if (!aRareProperty && bRareProperty)
                     return -1;
 
+                let aCaseSensitiveMatch = a.startsWith(prefix);
+                let bCaseSensitiveMatch = b.startsWith(prefix);
+                if (aCaseSensitiveMatch && !bCaseSensitiveMatch)
+                    return -1;
+                if (!aCaseSensitiveMatch && bCaseSensitiveMatch)
+                    return 1;
```

Comment:
This is where I preserved the cases the user has typed. Without these lines, upper case would always come first. 

Do you see a simpler way of doing this, BTW?

Reply:
oh i see.  this isn't so much about whether it's case sensitive, but about whether the user typed uppercase vs lowercase and preference based on that.  in that case this seems fine.  unfortunate that we're having to do _another_ two string comparisons each time we compare two strings, but i suppose this is the only way

Reply:
> unfortunate that we're having to do another two string comparisons each time we compare two strings, but i suppose this is the only way

If we want to do save as much work about comparing the strings while sorting, we could consider pre-computing the sorting criteria and make a tuple of data for each string, like this:
```js
// completions = ["string1", "string2", ...];

let completionSortCriteria = {};
for (let completion of completions) {
    let numericValue = isNaN(completion) ? Infinity : +completion;
    let isRareProperty = completion.startsWith("__") && completion.endsWith("__");
    let isNotExactMatch = !completion.startsWith(prefix);
    completionSortCriteria[completion] = [numericValue, isRareProperty, isNotExactMatch];
}

function compare(s, t)
{
    let a = completionSortCriteria[s];
    let b = completionSortCriteria[t];
    return a[0] - b[0] || a[1] - b[1] || a[2] - b[2] || s.extendedLocaleCompare(t);
}

completions.sort(compare);
```

Let me know if you think this would make a good fix.

Reply:
yeah i like that!

id probably use a `Map` instead of `{}`, and have the value be an object `{numericValue, isRareProperty, hasMatchingPrefix}`, but otherwise i am a fan :)

also, should this (perhaps preemptively) use `WI.CSSCompletions.getCompletionText` (tho we'd likely wanna move it to a more generic object)?